tests/basic.cpp

```diff
@@ -120,6 +120,24 @@ TEST_CASE("pretty_print") {
           "{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, ... size:14}");
   }
 
+  SECTION("container adapters") {
+    CHECK(pretty_print(std::priority_queue<int>()) == "{}");
+    CHECK(pretty_print(std::stack<int>({1, 2, 3})) == "{3, 2, 1}");
```

Comment:
Hm. Would it be less confusing to implement it in such a way that it outputs the elements in the correct order? Even if we can only `pop`, we could still collect a few elements into a temporary vector, and then output `{1, 2 3}` for this example, and sth. like `{..., 5, 6, 7, 8, 9, 10, 11, 12, 13, 14 (size: 14)}` for the example below.

Reply:
I initially thought of displaying elements as saved in the underlying container but doesn't that ruin the purpose of the adapter in the first place? I felt like it made more sense to display elements in correlation with the adapter functionality, hence the `pop` operation. What are your thoughts on this?

Reply:
I (believe) I understand. But the stack is also created with `std::stack<int>({1, 2, 3})`. And from a representation like `{1, 2, 3}`, I would expect the `pop` operation to remove the last element (3). Not the first. But I might be missing something here.

Reply:
I think I get what you are saying. That particular initialisation is directly initialising the underlying container (which is `std::vector` by default) . A more readable way would be like
```cpp
stack.push(1);
stack.push(2);
stack.push(3);
```
Please clarify whether this is your concern in which case I'll refactor the tests. Thanks for pointing this out!

Reply:
The initialization is not my concern. I'm just thinking: how do we represent a stack, which grows from "bottom"/first-pushed to "top"/last-pushed element? ("top" is an official term based on the API with a `top` function, but we can not "draw" it bottom to top, we can only go left to right)

We can either
(a) put the bottom on the left and the top on the right `{1, 2, 3}` or
(b) we can put the bottom on the right and the top on the left `{3, 2, 1}`.

I think it comes down to preference/convention. All I'm saying is that I find representation (a) with the topmost element on the right more natural, but you implemented (b), so presumably, you are of a different opinion? :smile:

Reply:
If you do it the (b) way, the expected output is `{1, 2, 3}`. This is because internally when elements are pushed to stack it is just pushed back to the end of the underlying container. When they are popped from stack it is just another `pop_back`. So `{1, 2, 3}` is equivalent to initialising the underlying container as `std::vector({1, 2, 3})` which by stack logic means a push of 1, followed by a push of 2, followed by a push of 3. When you pop from such a stack you expect to get elements in the order 3, 2, 1. 

This is why 
(a) `pretty_print(std::stack<int>({1, 2, 3}))` will display `{3, 2, 1}`
(b) `pretty_print(std::stack<int>({3, 2, 1}))` will display `{1, 2, 3}`

I hope the confusion is clarified :smile:

Reply:
Sorry, it took me a couple of passes through the messages to figure out the misunderstanding. I get it now :smile: .
Yes you are right, it sure does come down to preference. 
The reason I chose (b) over (a) is because the output stays consistent with the rest of the containers. However if you prefer the default implementation to be like (a), I will refactor as required.

Reply:
> because the output stays consistent with the rest of the containers

I don't get why?

Reply:
I see, because of the dots. I would still prefer variant (a), and would very much appreciate, if you could make the necessary changes.

Reply:
@sharkdp resolved.